src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -2137,6 +2137,25 @@ public void GenerateMethod(Method method, Class @class)
 
         private void GenerateOverloadCall(Function function)
         {
+            var needUnsafe = function.Parameters.Any(p => p.Kind == ParameterKind.Regular && p.Ignore
+                             && p.Type.IsPointerToPrimitiveType() && p.Usage == ParameterUsage.InOut && p.HasDefaultValue);
+            if(needUnsafe)
+            {
+                WriteLine("unsafe");
```

Comment:
Will move to the method signature.
Well, it is required because if there is a ref param with default value then it will make a native pointer with that value and then take its address to a managed type and pass the ref to managed type to the function call.